src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -556,27 +574,20 @@ protected override ImmutableArray<TypeParameterSymbol> ContainerTypeParameters
             get { return _owner.TypeParameters; }
         }
 
-        private TypeParameterConstraintClause GetTypeParameterConstraintClause(bool canIgnoreNullableContext)
-        {
-            return _owner.GetTypeParameterConstraintClause(canIgnoreNullableContext, this.Ordinal);
-        }
-
-        protected override TypeParameterBounds ResolveBounds(ConsList<TypeParameterSymbol> inProgress, bool canIgnoreNullableContext, DiagnosticBag diagnostics)
+        protected override TypeParameterBounds ResolveBounds(ConsList<TypeParameterSymbol> inProgress, DiagnosticBag diagnostics)
         {
-            var constraintClause = GetTypeParameterConstraintClause(canIgnoreNullableContext);
-            if (constraintClause.IsEmpty)
+            var constraintTypes = _owner.GetTypeParameterConstraintTypes(this.Ordinal);
+            if (constraintTypes.IsEmpty && GetConstraintKinds() == TypeParameterConstraintKind.None)
```

Comment:
> && GetConstraintKinds() == TypeParameterConstraintKind.None [](start = 39, length = 60)

I didn't follow. Why do we need the additional condition?

Reply:
> I didn't follow. Why do we need the additional condition?

Both conditions are equivalent to the original ```constraintClause.IsEmpty``` check. There is no intent to change when we call ResolveBounds

---
In reply to: [520174434](https://github.com/dotnet/roslyn/pull/49227#discussion_r520174434) [](ancestors = 520174434)